Consequences/ResultItem.cs

```diff
@@ -0,0 +1,21 @@
+﻿namespace Consequences;
+public class ResultItem<T>
+{
+  private string resultName;
+  private T result;
+
+  public ResultItem(string name, T res)
+  {
+    resultName = name;
+    result = res;
+  }
+
+  public string Name()
```

Comment:
There is some fancy syntactic sugar to beautify what you've done here. Auto Properties are what they're called. Automatically implements the backing field and the accessors. I'll show these tomorrow,.